File: GT-Models/src/main/java/com/DavideDalSanto/GTModels/Entities/UserExercise.java
package com.DavideDalSanto.GTModels.Entities;

import lombok.*;

import javax.persistence.*;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "user_exercises")
public class UserExercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "db_exercise_id")
    private Exercise exercise;

    private int reps;
    private int series;

//    @ManyToMany
//    @JoinTable(name = "workouts_user_exercises",
//            joinColumns = @JoinColumn(name = "workout_id"),
//            inverseJoinColumns = @JoinColumn(name = "user_exercises_id"))
//    private List<Workout> workouts = new ArrayList<>();

    //V2 fare string Note per segnare peso o PR
}

File: FE/GymTonic-FE/src/app/Models/user-exercise.ts
import { Exercise } from "./exercise";

export class UserExercise {
  id!:number;
  exercise:Exercise;
  reps:number;
  series:number;

  constructor(exercise:Exercise, reps:number, series:number) {
    this.exercise = exercise;
    this.reps = reps;
    this.series = series;
  }
}
